engines/python/setup/djl_python/properties_manager/tnx_properties.py

```diff
@@ -117,10 +125,26 @@ class TransformerNeuronXProperties(Properties):
     partition_schema: Optional[TnXModelSchema] = None
     all_reduce_dtype: Optional[TnXDtypeName] = None
     cast_logits_dtype: Optional[TnXDtypeName] = None
-    on_device_embedding: Optional[bool] = None
-    on_device_generation: Optional[Any] = Field(default_factory=dict)
+
+    max_model_len: Optional[int] = None
+    on_device_embedding: Optional[bool] = Field(
+        default_factory=lambda: get_env_or_default("NEURON_ON_DEVICE_EMBEDDING", None, convert_type=bool))
+    # TODO: on device generation could be bool, str or dictionary. Unify this.
+    on_device_generation: Optional[Any] = Field(
+        default_factory=lambda: get_env_or_default("NEURON_ON_DEV_GENERATION", None, convert_type=bool))
     shard_over_sequence: Optional[
-        bool] = None  # recommendation is true for batch size * sequence length > 16k
+        bool] = Field(
+        default_factory=lambda: get_env_or_default("NEURON_SHARD_OVER_SEQUENCE", False, convert_type=bool)
+    )  # recommendation is true for batch size * sequence length > 16k
+    compilation_worker_count: Optional[int] = Field(
+        default_factory=lambda: get_env_or_default("NEURON_COMPILATION_WORKER_COUNT", convert_type=int))
+    sequence_parallel: Optional[bool] = Field(
+        default_factory=lambda: get_env_or_default("NEURON_SEQUENCE_PARALLEL", True, bool))
+    multi_node: Optional[bool] = Field(
+        default_factory=lambda: get_env_or_default("NEURON_MULTI_NODE", False, convert_type=bool))
+    neuron_quant: Optional[bool] = Field(default_factory=lambda: get_env_or_default("NEURON_QUANT", False, bool))
```

Comment:
We may want to make this a quantization config instead to allow it to encompass its own set of configuration parameters in the long term.